src/WebAPI/ModelsAPI.cs

```diff
@@ -581,4 +582,83 @@ public static async Task<JObject> DoModelDownloadWS(Session session, WebSocket w
         }
         return null;
     }
+
+    [API.APIDescription("Returns the hash of a model, optionally  the hash from",
+        """
+        {
+            "success": true,
+            "model" {
+                "model": "My/Model/Filepath.safetensors",
+                "hash_sha256": "0xabc...",
+            },
+        }
+        """)]
+    public static async Task<JObject> GenerateModelHash(Session session,
+        [API.APIParameter("The full filepath of the model to load.")] string modelPath,
+        [API.APIParameter("The model's subtype, eg `Stable-Diffusion`, `LoRA`, etc.")] string modelSubtype = "Stable-Diffusion",
+        [API.APIParameter("If false, don't ever write the hash into the actual file itself. Set false if you will write additional metadata to this model.")] bool writeHashToDisk = true)
+    {
+        if (!Program.T2IModelSets.TryGetValue(modelSubtype, out T2IModelHandler modelHandler) && modelSubtype != "Wildcards")
+        {
+            return new JObject()
+            {
+                ["success"] = false,
```

Comment:
if `error` is present in an API result, that supercedes all other data, ie `"success": false` is redundant